R/module_nested_tabs.R

```diff
@@ -84,78 +78,69 @@ ui_nested_tabs.teal_modules <- function(id, modules, datasets, depth = 0L, is_mo
   )
 }
 
-#' @rdname module_nested_tabs
+#' @rdname module_teal_module
 #' @export
-ui_nested_tabs.teal_module <- function(id, modules, datasets, depth = 0L, is_module_specific = FALSE, progress = NULL) {
-  checkmate::assert_class(datasets, classes = "FilteredData")
+ui_teal_module.teal_module <- function(id, modules, depth = 0L) {
   ns <- NS(id)
-
-  if (!is.null(progress)) {
-    progress$inc(
-      amount = 1,
-      detail = sprintf("%s%%", round(progress$getValue() / progress$getMax(), 2L) * 100)
-    )
-  }
-
   args <- c(list(id = ns("module")), modules$ui_args)
-
-  teal_ui <- tags$div(
+  div(
     id = id,
     class = "teal_module",
     uiOutput(ns("data_reactive"), inline = TRUE),
     tagList(
       if (depth >= 2L) tags$div(style = "mt-6"),
-      do.call(modules$ui, args)
-    )
-  )
-
-  if (!is.null(modules$datanames) && is_module_specific) {
-    fluidRow(
-      column(width = 9, teal_ui, class = "teal_primary_col"),
-      column(
-        width = 3,
-        datasets$ui_filter_panel(ns("module_filter_panel")),
-        class = "teal_secondary_col"
+      fluidRow(
+        column(width = 9, do.call(modules$ui, args), class = "teal_primary_col"),
+        column(
+          width = 3,
+          ui_filter_panel(ns("module_filter_panel")),
+          class = "teal_secondary_col"
+        )
       )
     )
-  } else {
-    teal_ui
-  }
+  )
 }
 
-#' @rdname module_nested_tabs
-srv_nested_tabs <- function(id, datasets, modules, is_module_specific = FALSE,
+#' @rdname module_teal_module
+srv_teal_module <- function(id,
+                            data_rv,
+                            datasets,
+                            modules,
                             reporter = teal.reporter::Reporter$new()) {
   checkmate::assert_multi_class(modules, c("teal_modules", "teal_module"))
   checkmate::assert_class(reporter, "Reporter")
```

Comment:
add asserts on `id` and `data_rv`
we can leave `modules` to be handled by S3 dispatch